src/mon/FSCommands.cc

```diff
@@ -1245,6 +1242,125 @@ class RenameFilesystemHandler : public FileSystemCommandHandler
   Paxos *m_paxos;
 };
 
+class SwapFilesystemHandler : public FileSystemCommandHandler
+{
+  public:
+  explicit SwapFilesystemHandler(Paxos *paxos)
+    : FileSystemCommandHandler("fs swap"), m_paxos(paxos)
+  {
+  }
+
+  bool batched_propose() override {
+    return true;
+  }
+
+  int handle(
+      Monitor *mon,
+      FSMap& fsmap,
+      MonOpRequestRef op,
+      const cmdmap_t& cmdmap,
+      std::ostream &ss) override
+  {
+    ceph_assert(m_paxos->is_plugged());
+
+    // Check for confirmation flag
+    bool confirmation_flag = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", confirmation_flag);
+    if (!confirmation_flag) {
+      ss << "This is a potentially disruptive operation, client\'s cephx "
+	"credentials may need to be reauthorized to access the file systems "
+	"and its pools. Add --yes-i-really-mean-it if you are sure you wish "
+	"to continue.";
+      return -EPERM;
+    }
+
+    string fs1_name, fs2_name;
+    bool swap_flag;
+    cmd_getval(cmdmap, "fs1_name", fs1_name);
+    cmd_getval(cmdmap, "fs2_name", fs2_name);
+    cmd_getval(cmdmap, "swap_fscids", swap_flag);
+    auto fs1 = fsmap.get_filesystem(fs1_name);
+    auto fs2 = fsmap.get_filesystem(fs2_name);
+
+    // Check that CephFSs for both gives names exists
+    if (fs1 == nullptr or fs2 == nullptr) {
+      if (fs1 == nullptr and fs2 != nullptr) {
+	ss << "File system \"" << fs1_name << "\" doesn\'t exist on this "
+	      "Ceph cluster.";
+	return -ENOENT;
+      } else if (fs1 != nullptr and fs2 == nullptr) {
+	ss << "File system \"" << fs2_name << "\" doesn\'t exist on this "
+	      "Ceph cluster.";
+	return -ENOENT;
+      } else {
+	ss << "Neither file system \"" << fs1_name << "\" nor file "
+	      "system \"" << fs2_name << "\" exists on this Ceph cluster.";
+	return -ENOENT;
+      }
+    }
```

Comment:
this can be simplified to something like this

```
if(fs1 == nullptr && fs2 == nullptr) {
	ss << "Neither file system \"" << fs1_name << "\" nor file "
	      "system \"" << fs2_name << "\" exists on this Ceph cluster.";
	return -ENOENT;
	} else if(fs1 == nullptr) {
		ss << "File system \"" << fs1_name << "\" doesn\'t exist on this "
	      	"Ceph cluster.";
		return -ENOENT;
	} else if(fs2 == nullptr) {
		ss << "File system \"" << fs2_name << "\" doesn\'t exist on this "
	        "Ceph cluster.";
		return -ENOENT;
}
```

Reply:
I've deliberately added `if (fs1 == nullptr or fs2 == nullptr)` to avoid unnecessary checks in case neither is set to `nullptr`. So I've preferred optimization instead of simplification since it doesn't cause any fall in readability.

Reply:
skips traversing into if-else stmts if both exist

Reply:
Yes, that is the intended. The code is formatted as follows because it's a good programming practice -

```
if x is not ready:
  print(error)
  return
if y is not ready:
  print(error)
  return

final_action
```